### DISCUSSION TITLE
UPS CGI API is down

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3280] [Questions & Answers]
A couple of days UPS shut off their old CGI API that openmage also uses in the Mage_Usa module
<img width="642" alt="Screenshot 2023-05-27 alle 22 34 28" src="https://github.com/OpenMage/magento-lts/assets/909743/126e2b30-bdde-4996-8936-e9fede9ad4b3">
so now part of that module is completely useless and many websites are having big problems now.

Somebody says that the XML variant of the old API is still working but I can't verify directly right now:
<img width="640" alt="Screenshot 2023-05-27 alle 22 35 31" src="https://github.com/OpenMage/magento-lts/assets/909743/1bf74343-b038-4f59-ae97-cf6a2c750456">
but that one will also be discontinued sometime in 2024 (can't find the link where I've read it)

Magento 2 is having the same problem:
- https://community.magento.com/t5/Magento-2-x-Technical-Issues/UPS-not-showing-rates/m-p/512989
- https://github.com/magento/magento2/issues/37541
- https://github.com/magento/magento2/issues/37534

The new APIs are based on oauth, I've tried to crete a developer account on developer.ups.com but they want a real USA address and they don't accept my credit card (why do I have to provide a real CC for a developer account... I've no idea).

Does somebody have a UPS oauth ID/secret usable for testing?

**Comment by discountscott** [comment]
We have oauth credentials - as we use the UPS APIs internally (and recently began migrating from xml to REST). If you need us to test let me know.

I'll try and reach out to our UPS contact and see if it's possible to get a community developer account for testing.

**Reply by fballiano** [reply]
thanks you! I've managed to get an account after a few days, I'll try to write something for the rest API but it's a long work and I don't have that much time lately :-\

**Reply by boblewisvb** [reply]
Is there anyone working on implementing the UPS REST Api's?  On June 24, 2024 (~ 4 months) the current UPS module will cease to work.

**Reply by addison74** [reply]
Maybe we should spend some money and hire a programmer to make all the requested changes in time.

**Reply by fballiano** [reply]
exactly, if would be enough that a store sponsors the development of the module

**Reply by fballiano** [reply]
@discountscott @boblewisvb I need help with https://github.com/OpenMage/magento-lts/pull/3878
